WooCommerce/UITestsFoundation/Screens/Products/ProductSearchScreen.swift

```diff
@@ -0,0 +1,47 @@
+import ScreenObject
+import XCTest
+
+public final class ProductSearchScreen: ScreenObject {
+
+    private let searchTextFieldGetter: (XCUIApplication) -> XCUIElement = {
+        $0.descendants(matching: .other).element(matching: .any, identifier: "product-search-screen-search-field")
+    }
+
+    private var searchTextField: XCUIElement { searchTextFieldGetter(app) }
+    private var updatedSearchResultsID = "updated-search-results-table-view"
+    public static let defaultSearchResultsID = "default-search-results-table-view"
+
+    public init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetters: [ searchTextFieldGetter ],
+            app: app
+        )
+    }
+
+    public func enterSearchCriteria(text: String) throws -> Self {
+        searchTextField.tap()
+        searchTextField.enterText(text: text)
+        return self
+    }
+
+    @discardableResult
+    public func verifyNumberOfProductsOnSearchResults(expectedNumberOfProducts: Int, accessibilityId: String? = defaultSearchResultsID) throws -> Self {
+        let productTableView = app.tables.matching(identifier: accessibilityId!)
+        XCTAssert(productTableView.element.waitForExistence(timeout: 5), "Product list not displayed!")
+
+        let expectedCount = expectedNumberOfProducts
+        let actualCount = productTableView.cells.count
+
+        XCTAssertTrue(expectedCount == actualCount, "Expected \(expectedCount) products but found \(actualCount) instead!")
```

Comment:
Several comments here.

1. If `XCTAssertEqual` is used instead, we don't need to provide expected/actual values for the log message. XCTest will provide them automatically in case of a fail: `XCTAssertEqual failed: ("2") is not equal to ("4") - Products number is not expected!`
2. Thanks to ^ this ^, there's no longer need to keep `expectedCount` and `actualCount` as they are used only once:

```suggestion
        XCTAssertEqual(expectedNumber, productTableView.cells.count, "Products number is not expected!")
```

Reply:
Makes sense, `XCTAssertEqual` is better in this case. Updated in https://github.com/woocommerce/woocommerce-ios/pull/9276/commits/bbe389a4d63e8c4e793c02e3b34cbc15701ee9cf